### ISSUE TITLE
PyTorch script copies all the embeddings into GPU0 in a multi-gpu environment.

### ISSUE THREAD

**Issue by rohithkrn** [facebookresearch/dlrm#24]
In the below line `dlrm = dlrm.to(device)` all the tables are initially copied to gpu0 and then copied to other devices in `parallel_forward()` .
https://github.com/facebookresearch/dlrm/blob/e3af767980a2f61438104052c12ac75974a5e005/dlrm_s_pytorch.py#L656

This will be a problem when you pass in large embeddings and all of them do not fit in a single gpu. Ideally we should be able to split tables across gpus without the need to copy them to one gpu first. That is how the caffe2 script is doing it.

**Comment by mnaumovfb** [comment]
You are correct. The current PyTorch primitives which we used to build the parallel version of the code require us to make a copy of the entire model to gpu0. This is one of the reasons we state that we have a functionally correct parallel implementation, but that it is not performant.

In the future, we would like to adjust it to use different primitives to avoid this limitation, but we did not have a chance to do it, yet.
